tests/functional/placementapi_controller_test.go

```diff
@@ -843,6 +853,70 @@ var _ = Describe("PlacementAPI controller", func() {
 				ContainSubstring("[client]\nssl-ca=/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem\nssl=1"))
 		})
 	})
+
+	// Run MariaDBAccount suite tests.  these are pre-packaged ginkgo tests
+	// that exercise standard account create / update patterns that should be
+	// common to all controllers that ensure MariaDBAccount CRs.
+	mariadb_test.MariaDBAccountSuiteTests(
+		"Placememt",
+		// Populate test variables which will run inside of BeforeEach
+		func(harness *mariadb_test.MariaDBTestHarness) {
```

Comment:
As golang does not have keyword args these functions does not have a name here to signify their purpose and 
 their requirements. Would it be more readable to pass a struct with func fields to MariaDBAccountSuiteTests instead? Or does it violate some assumption within the ginkgo execution model itself?

Reply:
not sure?  I was just doing this whatever way I could get the funcs passed along, what benefits does a struct offer?   the thing about populating structs is that it seems you can leave fields omitted which is what I dont want here, every field is required.

Reply:
OK struct has names.   So.... make one struct and send to the function, how about this?  give this the thumbs up and I can change to this, I added a declare/run/go/???? method which takes the helper and timeout/interval


```
	// Run MariaDBAccount suite tests.  these are pre-packaged ginkgo tests
	// that exercise standard account create / update patterns that should be
	// common to all controllers that ensure MariaDBAccount CRs.
	mariadb_suite := &mariadb_test.MariaDBAccountSuite{
		name:          "Placement",
		namespace:     names.PlacementAPIName.Namespace,
		databaseName:  placement.DatabaseName,
		finalizerName: "PlacementAPI",
		generateCR: func(accountName types.NamespacedName) {
			// ...
		},
		updateAccount: func(newAccountName types.NamespacedName) {

			Eventually(func(g Gomega) {
				placementapi := GetPlacementAPI(names.PlacementAPIName)
				placementapi.Spec.DatabaseAccount = newAccountName.Name
				g.Expect(th.K8sClient.Update(ctx, placementapi)).Should(Succeed())
			}, timeout, interval).Should(Succeed())

		},
		deleteCR: func() {
			th.DeleteInstance(GetPlacementAPI(names.PlacementAPIName))
		},
	}
        // declareGinkgo() ?  run()?  doit() ?
	mariadb_suite.declareSuite(
		mariadb, timeout, interval,
	)

```

Reply:
on the optionality of the struct fields, that is a downside here yes, but you can add a runtime check for it.

Reply:
my soft preference is towards mariadb_suite.Run()

Reply:
I added additional "optional / run multiple times" suites to it as well so it now has `RunBasicSuite()` `RunURLAssertSuite(assertURL assertsURL)` `RunConfigHashSuite(getConfigHash getsConfigHash)` 

i think this makes it a lot easier to test everyone's ability to render expected URLs upfront/on change as well as being able to assert any number of confighashes that are expected to change